### PR TITLE
Feat/critical section requirement

### LINKED ISSUE #40: Add code requirement for critical section perimeter
NSR-10 C.11.11.1.2
ACI 318-19 22.6.4

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR closes #40

The perimeter is computed in the same way for ACI and NSR

image

image

tested cases:

image

### CODE REVIEW

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      module RC
+        module Footings
+          class PunchingCriticalSectionPerimeter
+            include DesignCodes::Utils::CodeRequirement
+            use_schema DesignCodes::Schemas::RC::Footings::PunchingCriticalSectionPerimeterSchema
+
+            def call
+              add_column_to_local_coordinates_system
+              add_perimeter_vertices_to_local_coordinates_system
+              edges_indexes = build_edges_indexes
```
Comment (ImMPrada) [inline lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb:13]: permeter_verties_relative-to-column

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+              ]
+            end
+
+            def perimeter_vertices_relative_to_column_location_with(way_1:, way_2:)
+              Vector[
+                way_1 * 0.5 * (column_section_length_1 + footing.effective_height),
+                way_2 * 0.5 * (column_section_length_2 + footing.effective_height),
+                0.0
+              ]
+            end
```
Comment (ImMPrada) [inline lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb:98]: permeter_verties_relative-to-column

File: lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+              relative_location = Engineering::Locations::Relative.new(
+                value_1: 0, value_2: 0, value_3: 0
+              )
+              relative_location.update_from_vector(vector)
+              local_coordinates_system.add_location(relative_location)
+            end
+
+            def relative_location_at_index(index)
+              local_coordinates_system.relative_locations[index]
+            end
```
Comment (ImMPrada) [inline lib/structuraid_core/design_codes/aci_318_19/rc/footings/punching_critical_section_perimeter.rb:110]: permeter_verties_relative-to-column

**Review by ImMPrada — COMMENTED** [review]
some notes

**Review by mapra99 — APPROVED** [review]

**Pull request merged**